src/napari/settings/_application.py

```diff
@@ -269,6 +272,27 @@ class ApplicationSettings(EventedModel):
         ),
     )
 
+    startup_script: str = Field(
+        default='',
+        title=trans._('Full path to a startup script'),
+        description=trans._(
+            'Path to a Python script that will be executed on napari startup. '
+            'This can be used to customize the behavior of napari or load specific plugins automatically.'
+        ),
+    )
+
+    def __setattr__(self, name: str, value: Any) -> None:
+        if name == 'startup_script' and value is not None:
+            # Ensure the script is a valid Python file
+            caller_frame = inspect.currentframe().f_back
+            if caller_frame is None or not caller_frame.f_globals.get(
+                '__name__', ''
+            ).startswith('napari.'):
+                raise ValueError(
+                    "The 'startup_script' setting can only be set by the napari application itself."
+                )
+        super().__setattr__(name, value)
+
```

Comment:
This code is protection from easily overwriting the value of `startup_script` from plugins.